lib/OntologyAPI/OntologyAPIImpl.py

```diff
@@ -0,0 +1,606 @@
+
+from OntologyAPI.utils import re_api, misc
+from pprint import pprint
+#END_HEADER
+
+
+class OntologyAPI:
+    '''
+    Module Name:
+    OntologyAPI
+
+    Module Description:
+    A KBase module: OntologyAPI
+    '''
+
+    ######## WARNING FOR GEVENT USERS ####### noqa
+    # Since asynchronous IO can lead to methods - even the same method -
+    # interrupting each other, you must be *very* careful when using global
+    # state. A method could easily clobber the state set by another while
+    # the latter method is running.
+    ######################################### noqa
+    VERSION = "0.3.3"
+    GIT_URL = "git@github.com:kbaseapps/ontology_api.git"
+    GIT_COMMIT_HASH = "e20d25d81d81722a5dd97adaf87d5cc083f36756"
+
+    #BEGIN_CLASS_HEADER
+    def validate_params(self, params, schema='default'):
+        _params = misc.validate_params(params, schema);
+        return _params
+    #END_CLASS_HEADER
+
+    # config contains contents of config file in a hash or None if it couldn't
+    # be found
+    def __init__(self, config):
+        #BEGIN_CONSTRUCTOR
+        self.shared_folder = config['scratch']
+        logging.basicConfig(format='%(created)s %(levelname)s: %(message)s',
+                            level=logging.INFO)
+        #END_CONSTRUCTOR
+        pass
+
+
+    def get_descendants(self, ctx, GenericParams):
+        """
+        Retrieve descendants of an ontology term by ID
+        :param GenericParams: instance of type "GenericParams" (Generic
+           Parameters id - required - ontology term id, such as "GO:0016209"
+           ts - optional - fetch documents with this active timestamp,
+           defaults to now ns - optional - ontology namespace to use,
+           defaults to "go" limit - optional - number of results to return
+           (defaults to 20) offset - optional - number of results to skip
+           (defaults to 0)) -> structure: parameter "id" of type "ID"
+           (Ontology term id, such as "GO:0000002"), parameter "ts" of Long,
+           parameter "ns" of String, parameter "limit" of Long, parameter
+           "offset" of Long
+        :returns: instance of type "GenericResults" (Generic results stats -
+           Query execution information from ArangoDB. results - array of
+           objects of results. ts - Timestamp used in the request ns -
+           Ontology namespace used in the request.) -> structure: parameter
+           "stats" of unspecified object, parameter "results" of list of
+           unspecified object, parameter "ts" of Long, parameter "ns" of
+           String
+        """
+        # ctx is the context object
+        # return variables are: returnVal
+        #BEGIN get_descendants
+        validated_params=self.validate_params(GenericParams);
+        results = re_api.query("get_descendants", validated_params)
+
+        returnVal={"stats": results["stats"], "results": results["results"], "ts": validated_params["ts"], "ns": validated_params["ns"]}
```

Comment:
Something like this would be a little more concise, if wanted
```
keys = {'stats', 'results', 'ts', 'ns'}
returnVal = {key: results[key] for key in keys}
```

Reply:
some are taken from `results` and some are taken from `validated_params`, so I will leave it as is.